tests/unittests/executor/test_executor.py

```diff
@@ -246,17 +247,38 @@ def test_nested_submit(backend):
             assert r.value == 35
 
 
-@pytest.mark.parametrize("backend", [multiprocess, thread])
-def test_nested_submit_failure(backend):
+def inc(a):
+    return a + 1
+
+
+def nested_pool():
+    data = [1, 2, 3, 4, 5, 6]
+    with multiprocessing.Pool(5) as p:
+        result = p.map_async(inc, data)
+        result.wait()
+        data = result.get()
+
+    return sum(data)
+
+
+@pytest.mark.parametrize("backend", backends)
+def test_nested_submit_pool(backend):
+    if backend is Dask:
+        pytest.skip("Dask does not support nesting")
+
     with backend(5) as executor:
+        futures = [executor.submit(nested_pool) for i in range(5)]
+
+        results = executor.async_get(futures, timeout=2)
 
-        if backend == multiprocess:
-            exception = NotImplementedError
-        elif backend == thread:
-            exception = TypeError
+        for r in results:
+            assert r.value == 27
 
-        with pytest.raises(exception):
-            [executor.submit(nested, executor) for i in range(5)]
+
+@pytest.mark.parametrize("backend", [multiprocess, thread])
```

Comment:
How is this one different than `test_nested_submit`?

Reply:
ah right, it used to be different because of the __setstate__ I removed it